src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItemCollection.cs

```
@@ -378,11 +376,11 @@ private bool IsValidIndex(int index)
     /// <summary>
     ///  Do proper cleanup of ownership, etc.
     /// </summary>
-    private void OnAfterRemove(ToolStripItem item)
+    private void OnAfterRemove(ToolStripItem? item)
```

all references of this method should be making sure `item `is not null. Looking at use case in `RemoveAt ` I would throw index out of range instead of propagating `null`.

---

There is an existing ` if (item is not null)` check in this method. We usually make the parameter nullable in such cases.

---

Right @gpetrou. In this case, that check is not needed given `private `use, we know all call sites checking for null. It is safe to change.

---

@dreddy-work I added a separate commit. Is this what you had in mind?

---

yes. thank you.